main/src/ca/uwaterloo/flix/language/phase/Verifier.scala

```diff
@@ -99,8 +99,31 @@ object Verifier {
           case _ => tpe // TODO: VERIFIER: Add rest
 
         }
+        case AtomicOp.Tag(sym) =>
+          root.enums.get(sym.enumSym) match {
+            case None => throw InternalCompilerException(s"Unknown enum sym: '$sym'", loc)
+            case Some(e) => e.cases.get(sym) match {
+              case None => throw InternalCompilerException(s"Unknown enum case sym: '$sym' of '${e.sym}'", loc)
+              case Some(caze) =>
+                val List(t) = ts
+                check(expected = caze.tpeDeprecated)(t, loc)
```

Comment:
note that these will be renamed tpe in the cleanup PR. The enum type is just `List$123` and the case type is the ground value of the tag `(Int32, String)` fx.